src/Symfony/Component/Serializer/Normalizer/TranslatableNormalizer.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Contracts\Translation\TranslatableInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+final class TranslatableNormalizer implements NormalizerInterface
+{
+    public const NORMALIZATION_LOCALE_KEY = 'translatable_normalization_locale';
+
+    private array $defaultContext = [
+        self::NORMALIZATION_LOCALE_KEY => null,
+    ];
+
+    public function __construct(
+        array $defaultContext = [],
+        private readonly ?TranslatorInterface $translator = null,
+    ) {
+        $this->defaultContext = array_merge($this->defaultContext, $defaultContext);
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function normalize(mixed $object, string $format = null, array $context = []): string
+    {
+        if (!$object instanceof TranslatableInterface) {
+            throw new InvalidArgumentException(sprintf('The object must implement the "%s".', TranslatableInterface::class));
+        }
+
+        return $object->trans($this->translator, $context[self::NORMALIZATION_LOCALE_KEY] ?? $this->defaultContext[self::NORMALIZATION_LOCALE_KEY]);
+    }
+
+    public function supportsNormalization(mixed $data, string $format = null, array $context = []): bool
+    {
+        return null !== $this->translator && $data instanceof TranslatableInterface;
+    }
+
+    public function getSupportedTypes(?string $format): array
+    {
+        return [TranslatableInterface::class => true];
```

Comment:
With the current implementation, this method should return an empty array of no translator is injected. But see my other comment.